`open = true` failed to build for me atleast.

<details><summary>Details</summary>
<p>

```nix

nvidia-open> make[1]: *** [Makefile:203: _out/Linux_x86_64/g_console_mem_nvoc.o] Error 1
nvidia-open> 1 error generated.
nvidia-open> make[1]: *** [Makefile:203: _out/Linux_x86_64/g_client_nvoc.o] Error 1
nvidia-open> make[1]: *** [Makefile:203: _out/Linux_x86_64/g_conf_compute_api_nvoc.o] Error 1
nvidia-open> make[1]: *** [Makefile:203: _out/Linux_x86_64/g_compute_instance_subscription_nvoc.o] Error 1
nvidia-open> 1 error generated.
nvidia-open> make[1]: *** [Makefile:203: _out/Linux_x86_64/g_conf_compute_nvoc.o] Error 1
nvidia-open> make[1]: *** [Makefile:203: _out/Linux_x86_64/g_client_resource_nvoc.o] Error 1
nvidia-open> generated/g_crashcat_engine_nvoc.c:2:10: fatal error: 'nvoc/runtime.h' file not found
nvidia-open>     2 | #include "nvoc/runtime.h"
nvidia-open>       |          ^~~~~~~~~~~~~~~~
nvidia-open> make[1]: *** [Makefile:203: _out/Linux_x86_64/g_context_dma_nvoc.o] Error 1
nvidia-open> 1 error generated.
nvidia-open> make[1]: *** [Makefile:203: _out/Linux_x86_64/g_crashcat_engine_nvoc.o] Error 1
nvidia-open> 1 error generated.
nvidia-open> make[1]: *** [Makefile:203: _out/Linux_x86_64/g_bindata.o] Error 1
nvidia-open> make[1]: Leaving directory '/build/source/src/nvidia'
nvidia-open> make: *** [Makefile:34: src/nvidia/_out/Linux_x86_64/nv-kernel.o] Error 2
error: builder for '/nix/store/nlyawfgxkjjm2v0cyq8yz2ksz41axbdj-nvidia-open-6.10.0-560.28.03.drv' failed with exit code 2;
       last 40 log lines:
       >       |          ^~~~~~~~~~~~~~~~
       > generated/g_client_resource_nvoc.c:2:10: fatal error: 'nvoc/runtime.h' file not found
       >     2 | #include "nvoc/runtime.h"
       >       |          ^~~~~~~~~~~~~~~~
       > 1 error generated.
       > 1 error generated.
       > make[1]: *** [Makefile:203: _out/Linux_x86_64/g_ccsl_nvoc.o] Error 1
       > make[1]: *** [Makefile:203: _out/Linux_x86_64/g_chips2halspec_nvoc.o] Error 1
       > 1 error generated.
       > make[1]: *** [Makefile:203: _out/Linux_x86_64/g_all_dcl_pb.o] Error 1
       > 1 error generated.
       > make[1]: Leaving directory '/build/source/src/nvidia-modeset'
       > make[1]: *** [Makefile:203: _out/Linux_x86_64/g_chipset_nvoc.o] Error 1
       > make: *** [Makefile:46: src/nvidia-modeset/_out/Linux_x86_64/nv-modeset-kernel.o] Error 2
       > make: *** Waiting for unfinished jobs....
       > make[1]: *** [Makefile:203: _out/Linux_x86_64/g_channel_descendant_nvoc.o] Error 1
       > make[1]: *** [Makefile:203: _out/Linux_x86_64/g_binary_api_nvoc.o] Error 1
       > make[1]: *** [Makefile:203: _out/Linux_x86_64/g_ce_utils_nvoc.o] Error 1
       > 1 error generated.
       > generated/g_context_dma_nvoc.c:2:10: fatal error: 'nvoc/runtime.h' file not found
       >     2 | #include "nvoc/runtime.h"
       >       |          ^~~~~~~~~~~~~~~~
       > make[1]: *** [Makefile:203: _out/Linux_x86_64/g_console_mem_nvoc.o] Error 1
       > 1 error generated.
       > make[1]: *** [Makefile:203: _out/Linux_x86_64/g_client_nvoc.o] Error 1
       > make[1]: *** [Makefile:203: _out/Linux_x86_64/g_conf_compute_api_nvoc.o] Error 1
       > make[1]: *** [Makefile:203: _out/Linux_x86_64/g_compute_instance_subscription_nvoc.o] Error 1
       > 1 error generated.
       > make[1]: *** [Makefile:203: _out/Linux_x86_64/g_conf_compute_nvoc.o] Error 1
       > make[1]: *** [Makefile:203: _out/Linux_x86_64/g_client_resource_nvoc.o] Error 1
       > generated/g_crashcat_engine_nvoc.c:2:10: fatal error: 'nvoc/runtime.h' file not found
       >     2 | #include "nvoc/runtime.h"
       >       |          ^~~~~~~~~~~~~~~~
       > make[1]: *** [Makefile:203: _out/Linux_x86_64/g_context_dma_nvoc.o] Error 1
       > 1 error generated.
       > make[1]: *** [Makefile:203: _out/Linux_x86_64/g_crashcat_engine_nvoc.o] Error 1
       > 1 error generated.
       > make[1]: *** [Makefile:203: _out/Linux_x86_64/g_bindata.o] Error 1
       > make[1]: Leaving directory '/build/source/src/nvidia'
       > make: *** [Makefile:34: src/nvidia/_out/Linux_x86_64/nv-kernel.o] Error 2
       For full logs, run 'nix log /nix/store/nlyawfgxkjjm2v0cyq8yz2ksz41axbdj-nvidia-open-6.10.0-560.28.03.drv'.
       
```

</p>
</details>

--- a/nixos/modules/hardware/video/nvidia.nix
+++ b/nixos/modules/hardware/video/nvidia.nix
@@ -253,7 +253,9 @@ in
 
       open = lib.mkEnableOption ''
         the open source NVIDIA kernel module
-      '';
+      '' // {
+        defaultText = lib.literalExpression ''lib.versionAtLeast config.hardware.nvidia.package.version "560"'';
+      };
     };
   };
 
@@ -302,6 +304,8 @@ in
             extraPackages32 = [ nvidia_x11.lib32 ];
           };
           environment.systemPackages = [ nvidia_x11.bin ];
+
+          hardware.nvidia.open = lib.mkDefault (lib.versionAtLeast nvidia_x11.version "560");
         })
 
         # X11

